### PR TITLE
[AOT] Return module list from AotExecutorFactory

### PR DESCRIPTION
**Pull request by kparzysz-quic**
Add a function `module_list` to `AotExecutorFactory` that returns `Array<String>` containing names of all modules. At the moment there will be only one entry in that list, most commonly "default".

cc @Mousius

### CODE REVIEW

File: tests/python/relay/aot/test_cpp_aot.py [hunk 1]
```diff
@@ -145,6 +145,26 @@ def test_mobilenet(enable_usmp, target_kind):
...
+    temp_dir = tvm.contrib.utils.TempDirectory()
+    test_so_path = temp_dir / "test.so"
+    mod.export_library(test_so_path, cc="gcc", options=["-std=c11"])
+    loaded_mod = tvm.runtime.load_module(test_so_path)
+    module_list = loaded_mod.get_function("module_list")
+    # At the moment, relay produces a single module with the name "default".
+    names_expected = ["default"]
+    names_obtained = sorted(module_list())
+    assert len(names_obtained) == len(names_expected)
+    assert all([x == y for (x, y) in zip(names_obtained, names_expected)])
```
Comment (areusch) [inline tests/python/relay/aot/test_cpp_aot.py:165]: i think these two could just be 
`assert names_expected = list(sorted(module_list())`

and then it'd be easier to debug if it broke

File: tests/python/relay/aot/test_cpp_aot.py [hunk 2]
```diff
@@ -145,6 +145,26 @@ def test_mobilenet(enable_usmp, target_kind):
     assert (runner.get_output(0).asnumpy() == list(ref_outputs.values())[0]).all()
 
 
+def test_module_list():
+    x = tvm.relay.var("x", tvm.relay.TensorType([1], dtype="float32"))
+    expr = tvm.relay.add(x, tvm.relay.Constant(tvm.nd.array(np.array([1], dtype="float32"))))
+    mod = tvm.relay.build(
```
Comment (areusch) [inline tests/python/relay/aot/test_cpp_aot.py:151]: nit: could pass mod_name = 'foo' here and then test that to verify the knob works

File: src/runtime/aot_executor/aot_executor_factory.cc [hunk 3]
```diff
@@ -52,6 +52,11 @@ PackedFunc AotExecutorFactory::GetFunction(
       }
       *rv = this->ExecutorCreate(devices);
     });
+  } else if (name == "module_list") {
```
Comment (areusch) [inline src/runtime/aot_executor/aot_executor_factory.cc:55]: to follow convention of other functions here, shall we say list_module_names?

**Review by areusch — COMMENTED** [review]
thanks @kparzysz-quic , this would indeed be helpful! left a few thoughts

cc @alanmacd @tqchen @csullivan @mehrdadh @comaniac

File: tests/python/relay/aot/test_cpp_aot.py (hunk 1 above)
Comment (kparzysz-quic) [reply tests/python/relay/aot/test_cpp_aot.py]: Done.

File: tests/python/relay/aot/test_cpp_aot.py (hunk 2 above)
Comment (kparzysz-quic) [reply tests/python/relay/aot/test_cpp_aot.py:151]: Done.

File: src/runtime/aot_executor/aot_executor_factory.cc (hunk 3 above)
Comment (kparzysz-quic) [reply src/runtime/aot_executor/aot_executor_factory.cc]: Done.

**Review by csullivan — APPROVED** [review]

**Pull request merged**